### ISSUE TITLE
CORS failure in k8s env.

### ISSUE THREAD

**Issue by vVv-AA** [Apicurio/apicurio-registry#3493]
## Description

**Registry
Version**: latest
**Persistence type**: sql

Getting error when trying to create an artifact from UI. 
CORS failure. Because of http vs https mismatch.

Tried adding env var
```
"CORS_ALLOWED_ORIGINS": "http://localhost:8080,http://<redacted>,https://<redacted>"
```

### Environment
Kubernetes v1.22.X. 

### Steps to Reproduce

Deployed registry in kubernetes cluster.
Setup ingress
```
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: apicurio-webserver-ingress
  annotations:
    nginx.ingress.kubernetes.io/ssl-redirect: "false"
    kubernetes.io/ingress.class: "nginx-internal"
spec:
  tls:
    - hosts:
        - <redacted>
      secretName: wildcard-tls-dev
  rules:
    - host: <redacted>
      http:
        paths:
          - backend:
              service:
                name: apicurio-webserver-service
                port:
                  number: 8080
            pathType: Prefix
            path: /
```
Open ui in web `https://<redacted>/ui`

### Expected vs Actual Behaviour

Artifact should get created. After adding CORS_ALLOWED_ORIGINS.
Is there a better way around this?

### Logs

```2023-07-07 06:54:31 TRACE <> [io.quarkus.scheduler.runtime.SimpleScheduler] (pool-9-thread-1) Check triggers at 2023-07-07T06:54:31.000247648Z[UTC]

2023-07-07 06:54:31 TRACE <> [null] (vert.x-eventloop-thread-6) Router: 1928697992 accepting request POST[ http://<redacted>/apis/registry/v2/groups/default/artifacts](http://<redacted>/apis/registry/v2/groups/default/artifacts)

2023-07-07 06:54:31 TRACE <> [null] (vert.x-eventloop-thread-6) Calling the  handler

2023-07-07 06:54:31 DEBUG <> [io.quarkus.vertx.http.runtime.cors.CORSFilter] (vert.x-eventloop-thread-6) Same origin check has failed, the origin is not a substring of the request URI. Request URI:[ http://<redacted>/apis/registry/v2/groups/default/artifacts](http://<redacted>/apis/registry/v2/groups/default/artifacts), origin: https://<redacted>/

2023-07-07 06:54:31 DEBUG <> [io.quarkus.vertx.http.runtime.cors.CORSFilter] (vert.x-eventloop-thread-6) Origin is not allowed```

**Comment by carlesarnal** [comment]
I wouldn't say this is a bug. You should not mix http and https, the error you're getting is kind of expected. The application should be deployed using https (http is fine for development purposes, but should not be used beyond that) and the UI will make https requests to the backend.

**Comment by vVv-AA** [comment]
Hi,

I have deployed using the prod profile and I am hitting from browser using https.
We do tls termination using the ingress controller. Are you saying this kind of setup will not work?
If not, is there any suggested way to handle such scenario?

I raised it as a bug adding http:// and https:// endpoints in the CORS_ALLOWED_ORIGINS env var did not work. 
If configuring the env var would work then it would solve the issue.
